File: examples/test_with_open.py
#!/usr/bin/env python3
"""
Simple test for with open() support
"""

import argparse


def main():
    parser = argparse.ArgumentParser(description="Test with open()")
    parser.add_argument("--write", action="store_true", help="Write to file")
    parser.add_argument("--read", action="store_true", help="Read from file")
    args = parser.parse_args()

    if args.write:
        with open("test.txt", "w") as f:
            f.write("Hello from Rust!\n")
        print("Written to test.txt")

    if args.read:
        with open("test.txt") as f:
            content = f.read()
        print(f"Read: {content}")


if __name__ == "__main__":
    main()

File: examples/test_with_open.rs
use clap::Parser;
use std::io::Read;
use std::io::Write;
#[derive(clap::Parser)]
#[command(about = "Test with open()")]
struct Args {
    #[arg(long)]
    #[arg(action = clap::ArgAction::SetTrue)]
    #[doc = "Write to file"]
    write: bool,
    #[arg(long)]
    #[arg(action = clap::ArgAction::SetTrue)]
    #[doc = "Read from file"]
    read: bool,
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn main() -> Result<(), std::io::Error> {
    let args = Args::parse();
    if args.write {
        let mut f = std::fs::File::create("test.txt".to_string())?;
        f.write_all("Hello from Rust!\n".to_string().as_bytes())
            .unwrap();
        println!("{}", "Written to test.txt");
    }
    if args.read {
        let mut f = std::fs::File::open("test.txt".to_string())?;
        let content = {
            let mut content = String::new();
            f.read_to_string(&mut content)?;
            content
        };
        println!("{}", format!("Read: {:?}", content));
    }
    Ok(())
}
